File: eKnjigaDesktopUI/lib/models/review.dart
class Review {
  final int id;
  final double rating;
  final DateTime createdAt;
  final String? bookTitle;
  final String? userFullName;

  Review({
    required this.id,
    required this.rating,
    required this.createdAt,
    this.bookTitle,
    this.userFullName,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      rating: json['rating'],
      createdAt: DateTime.parse(json['createdAt']),
      bookTitle: json['book']?['title'],
      userFullName: json['user'] != null
          ? "${json['user']['firstName']} ${json['user']['lastName']}"
          : null,
    );
  }
}

File: eKnjiga/eKnjiga.Services/Database/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eKnjiga.Services.Database
{
    public class Review
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public double Rating { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Foreign key for Book
        public int BookId { get; set; }
        [ForeignKey("BookId")]
        public Book Book { get; set; }

        // Foreign key for User
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }
    }
} 